sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -379,8 +399,8 @@ def _conn_string_adodbapi(self, db_key, instance=None, conn_key=None, db_name=No
             _, host, username, password, database, _ = self._get_access_info(instance, db_key, db_name)
         elif conn_key:
             _, host, username, password, database, _ = conn_key.split(":")
-        conn_str = 'Provider=SQLOLEDB;Data Source=%s;Initial Catalog=%s;' \
-            % (host, database)
+        conn_str = 'Provider=%s;Data Source=%s;Initial Catalog=%s;' \
+            % (self._get_adoprovider(instance), host, database)
```

Comment:
`_conn_string_adodbapi` states that it's possible for `instance` to be `None`. There'll be a runtime error on L333 if `None` is passed through.